README.md

```diff
@@ -1,76 +1,124 @@
+### Step 4:  Configure the sample to use your Azure AD tenant
 
-```
+In the steps below, ClientID is the same as Application ID or AppId.
 
-### Step 6: Package and then run the `public-client-adal4j-sample-jar-with-dependencies.jar ` file.
+#### Configure the app project
+
+1. Open the `src\main\java\PublicClient.java` file
+1. Find the app key `private final static String CLIENT_ID` and replace the existing value with the application ID (clientId) of the `Native-Headless-Application` application copied from the Azure portal.
+
+### Step 5: Run the sample
 
 From your shell or command line:
 
-* `$ mvn package`
+- `$ mvn package`
 
 This will generate a `public-client-adal4j-sample-jar-with-dependencies.jar` file in your /targets directory. Run this using your Java executable like below:
 
-* `$ java -jar public-client-adal4j-sample-jar-with-dependencies.jar`
-
+- `$ java -jar public-client-adal4j-sample-jar-with-dependencies.jar`
 
 ### You're done!
 
-Your command line interface should prompt you for the username and password and then access the Graph API to retreive your user information.
+Your command line interface should prompt you for the username and password and then access the Microsoft Graph API to retrieve your user information.
+
+## Community Help and Support
+
+Use [Stack Overflow](http://stackoverflow.com/questions/tagged/adal) to get support from the community.
+Ask your questions on Stack Overflow first and browse existing issues to see if someone has asked your question before.
+Make sure that your questions or comments are tagged with [`adal` `Java`].
+
+If you find a bug in the sample, please raise the issue on [GitHub Issues](https://github.com/Azure-Samples/active-directory-java-native-headless/issues).
+
+To provide a recommendation, visit the following [User Voice page](https://feedback.azure.com/forums/169401-azure-active-directory).
+
+## Contributing
+
+If you'd like to contribute to this sample, see CONTRIBUTING.md
 
-### Acknowledgements
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information, see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
 
-We would like to acknowledge the folks who own/contribute to the following projects for their support of Azure Active Directory and their libraries that were used to build this sample. In places where we forked these libraries to add additional functionality, we ensured that the chain of forking remains intact so you can navigate back to the original package. Working with such great partners in the open source community clearly illustrates what open collaboration can accomplish. Thank you!
+## More information
 
+For more information, see ADAL4J [conceptual documentation](https://github.com/AzureAD/azure-activedirectory-library-for-java/wiki)
```

Comment:
Do you have a link on the specific flow with username/password in the wiki? 
if yo don't you might want to create it? (get inspired by the .NET one: [Acquiring tokens with username and password](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/Acquiring-tokens-with-username-and-password)

Reply:
Information on username/password is in: https://github.com/AzureAD/azure-activedirectory-library-for-java/wiki/Acquiring-Tokens-with-username-and-password